app.rb

```diff
@@ -16,84 +16,88 @@ def clear_popolo
-            slug: l.slug,
-          }
-          terms = {}
+        end
 
-          l.popolo.events.where(classification: "legislative period").each do |lp|
-            terms[lp.id] = {
-              overall: Hash.new(0),
-              parties: Hash.new { | hash, key | hash[key] = Hash.new(0) },
-            }
-          end
+        person_memberships = Hash.new { | hash, key | hash[key] = [] }
+        l.popolo.memberships.each do |m|
+          person_memberships[m.person_id].push(m)
+        end
 
-          person_memberships = Hash.new { | hash, key | hash[key] = [] }
-          l.popolo.memberships.each do |m|
-            person_memberships[m.person_id].push(m)
+        l.popolo.persons.each do |p|
+          gender = p.gender || 'unknown'
+          if not totals_seen[p.id]
+              totals[:overall][gender] += 1
+              totals[:overall][:total] += 1
+              totals_seen[p.id] = gender
+          elsif totals_seen[p.id] != gender and gender != 'unknown'
+              totals[:overall][gender] += 1
+              totals[:overall][totals_seen[p.id]] -= 1
+              totals_seen[p.id] = gender
           end
-
-          l.popolo.persons.each do |p|
-            gender = p.gender || 'unknown'
-            if not totals_seen[p.id]
-                totals[:overall][gender] += 1
-                totals[:overall][:total] += 1
-                totals_seen[p.id] = gender
-            elsif totals_seen[p.id] != gender and gender != 'unknown'
-                totals[:overall][gender] += 1
-                totals[:overall][totals_seen[p.id]] -= 1
-                totals_seen[p.id] = gender
-            end
-            person_memberships[p.id].each do |m|
-              term_id = m.document[:legislative_period_id]
-              terms[term_id][:overall][gender] += 1
-              terms[term_id][:overall][:total] += 1
-              party = m.on_behalf_of_id
-              terms[term_id][:parties][party][gender] += 1
-              terms[term_id][:parties][party][:total] += 1
-              party_person = party + ':' + p.id
-              if not totals_seen[party_person]
-                  totals[:parties][party][gender] += 1
-                  totals[:parties][party][:total] += 1
-                  totals_seen[party_person] = gender
-              elsif totals_seen[party_person] != gender and gender != 'unknown'
-                  totals[:overall][party][gender] += 1
-                  totals[:overall][party][totals_seen[party_person]] -= 1
-                  totals_seen[party_person] = gender
-              end
+          person_memberships[p.id].each do |m|
+            term_id = m.document[:legislative_period_id]
+            terms[term_id][:overall][gender] += 1
+            terms[term_id][:overall][:total] += 1
+            party = m.on_behalf_of_id
+            terms[term_id][:parties][party][gender] += 1
+            terms[term_id][:parties][party][:total] += 1
+            party_person = party + ':' + p.id
```

Comment:
I'm not really following what's going on here, but this doesn't seem like the way to do it…